src/depthai_wrapper.cpp

```diff
@@ -0,0 +1,150 @@
+    pipeline_ = create_pipeline(config_str);
+    config_json_ = nlohmann::json::parse(config_str);
+
+    // extract resolution of the frames and create cv::Mat placeholders
+    set_resolution();
+    create_frame_holders();
+}
+
+void DepthAI::set_resolution()
+{
+    // if camera is defined check if resolution of stereo or rgb is defined. else set to default
+    if (config_json_.contains("camera")) 
+    {
+        auto& camera_conf_obj = config_json_.at("camera");
+        /*
+         * if rgb res is set in config_json_ fetch it to set 
+         * the height of the image else set the default
+         */
+        if (camera_conf_obj.contains("rgb")) 
+        {
+            auto& rgb_camera_conf_obj = camera_conf_obj.at("rgb");
+            rgb_height_ = rgb_camera_conf_obj.at("resolution_h").get<int32_t>();
+            // _rgb_width = height_to_width_map_[_rgb_height];
+        } else 
+        {
+            rgb_height_ = default_rgb_height_;
+            rgb_width_ = default_rgb_width_;
+        }
+        /*
+         * if stereo camera res is set in config_json_ fetch it to set 
+         * the height of the image else set the default
+         */
+        if (camera_conf_obj.contains("mono")) 
+        {
+            auto& mono_camera_conf_obj = camera_conf_obj.at("mono");
+            mono_height_ = mono_camera_conf_obj.at("resolution_h").get<int32_t>();
+            // auto it = height_to_width_map_.find(mono_height_);
+            // mono_width_ = it->second;
+            mono_width_ = height_to_width_map_.at(mono_height_);
+        } else 
+        {
+            mono_height_ = default_mono_height_;
+            mono_width_ = default_mono_width_;
+        }
+    } else 
+    {
+        rgb_height_ = default_rgb_height_;
+        rgb_width_ = default_rgb_width_;
+        mono_height_ = default_mono_height_;
+        mono_width_ = default_mono_width_;
+    }
+}
+
+void DepthAI::create_frame_holders()
+{
+    // creating place holders for all the streams enabled in config
+    if (config_json_.contains("streams")) 
+    {
+        for (const auto& it : config_json_.at("streams")) 
+        {
+            if (it.is_string()) 
+            {
+                if (it.get<std::string>() == "metaout")
+                    continue;
+                if (it.get<std::string>() == "previewout") 
+                {
+                    CV_mat_ptr img = std::make_shared<cv::Mat>(rgb_height_, rgb_width_, CV_8UC3);
+                    image_stream_holder["previewout"] = img;
+                }
+            } else 
+            {
+                const auto& name = it.at("name").get<std::string>();
+                if (name == "previewout") 
+                {
+                    CV_mat_ptr img = std::make_shared<cv::Mat>(rgb_height_, rgb_width_, CV_8UC3);
+                    image_stream_holder["previewout"] = img;
+                } else if (name == "depth") 
+                {
```

Comment:
```suggestion
                } 
                else if (name == "depth") 
                {
```

Reply:
I had fixed it. formatting changed it again.